client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -88,11 +93,13 @@
     private final ExecutorService retransmitPool;
     private final Pinger pinger;
     private final DelegationTokenProvider tokenProvider;
+    private final ConnectionPool connectionPool;
     
     EventStreamWriterImpl(Stream stream, String writerId, Controller controller, SegmentOutputStreamFactory outputStreamFactory,
                           Serializer<Type> serializer, EventWriterConfig config, ExecutorService retransmitPool,
-                          ScheduledExecutorService internalExecutor) {
+                          ScheduledExecutorService internalExecutor, ConnectionPool connectionPool) {
         this.writerId = writerId;
+        this.connectionPool = connectionPool;
```

Comment:
We should validate that this not null. We use it blindly below.

Reply:
In the non-large-event unit tests, it is null.